### ISSUE TITLE
Warning when building doc after install in editable mode: more pachaging??

### ISSUE THREAD

**Issue by Remi-Gau** [bids-standard/pybids-reports#43]
```
pip install -e .[dev]
cd docs
make html
```

<details>

 <summary>error</summary>

 <pre>
Running Sphinx v7.0.1
loading pickled environment... done
myst v2.0.0: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=set(), disable_syntax=[], all_links_external=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, fence_as_directive=set(), number_code_blocks=[], title_to_header=False, heading_anchors=0, heading_slug_func=None, html_meta={}, footnote_transition=True, words_per_minute=200, substitutions={}, linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', enable_checkboxes=False, suppress_warnings=[], highlight_code_blocks=True)
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 0 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100%] reports                                                                                           
WARNING: autodoc: failed to import module 'reports' from module 'bids.ext'; the following exception was raised:
No module named 'bids.ext.reports'
WARNING: autodoc: failed to import module 'reports.parameters' from module 'bids.ext'; the following exception was raised:
No module named 'bids.ext.reports'
WARNING: autodoc: failed to import module 'reports.parsing' from module 'bids.ext'; the following exception was raised:
No module named 'bids.ext.reports'
WARNING: autodoc: failed to import module 'reports.report' from module 'bids.ext'; the following exception was raised:
No module named 'bids.ext.reports'
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/remi/github/bids/pybids-reports/docs/source/modules.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] reports                                                                                            
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 5 warnings.

 The HTML pages are in build/html.
 </pre>

</details>

compared to doc build when using `pip install .[dev]`

<details>

 <summary>error</summary>

 <pre>

Running Sphinx v7.0.1
loading pickled environment... done
myst v2.0.0: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=set(), disable_syntax=[], all_links_external=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, fence_as_directive=set(), number_code_blocks=[], title_to_header=False, heading_anchors=0, heading_slug_func=None, html_meta={}, footnote_transition=True, words_per_minute=200, substitutions={}, linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', enable_checkboxes=False, suppress_warnings=[], highlight_code_blocks=True)
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 0 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100%] reports                                                                                           
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/remi/github/bids/pybids-reports/docs/source/modules.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] reports                                                                                            
generating indices... genindex py-modindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in build/html.

 </pre>

</details>

**Comment by effigies** [comment]
Yes, I noticed an issue with editable installs yesterday but didn't have time to debug.

**Comment by Remi-Gau** [comment]
no worries: not urgent

**Comment by effigies** [comment]
I have a demo repository here: https://github.com/effigies/hatchling-namespace-demo

It works there. I'm not 100% on why things aren't working here. It may be the mismatch between the distribution name `pybids` and the module name `bids`.

**Comment by tsalo** [comment]
I tried it out locally with an editable install today and didn't get any warnings (my PR fixed the "document not in doctree" issue).

**Comment by Remi-Gau** [comment]
That'd be nice

**Comment by Remi-Gau** [comment]
now getting an error:


```
Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/home/remi/miniconda3/lib/python3.11/site-packages/sphinx/config.py", line 354, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/bids/pybids-reports/docs/source/conf.py", line 22, in <module>
    import bids.ext.reports  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'bids.ext.reports'
```
